app/org/sagebionetworks/bridge/crypto/CmsEncryptorCacheLoader.java

```diff
@@ -0,0 +1,51 @@
+package org.sagebionetworks.bridge.crypto;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.security.PrivateKey;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+
+import com.google.common.cache.CacheLoader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.config.BridgeConfigFactory;
+import org.sagebionetworks.bridge.s3.S3Helper;
+
+/**
+ * This is the cache loader that supports loading CMS encryptors on demand, keyed by the study ID. If the study's
+ * encryptor is already in the cache, this returns that encryptor. If it isn't, this study will pull the PEM fils for
+ * the cert and private key from the configured S3 bucket and construct an encryptor using those encryption materials.
+ */
+@Component
+public class CmsEncryptorCacheLoader extends CacheLoader<String, CmsEncryptor> {
+    public static final String CERT_BUCKET = BridgeConfigFactory.getConfig().getProperty("upload.cms.cert.bucket");
+    private static final String PEM_FILENAME_FORMAT = "%s.pem";
+    public static final String PRIV_KEY_BUCKET = BridgeConfigFactory.getConfig().getProperty(
+            "upload.cms.priv.bucket");
+
+    private S3Helper s3Helper;
+
+    /** S3 helper, configured by Spring. */
+    @Autowired
+    public void setS3Helper(S3Helper s3Helper) {
+        this.s3Helper = s3Helper;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public CmsEncryptor load(@Nonnull String study) throws CertificateEncodingException, IOException {
```

Comment:
Just to be super clear and consistent, perhaps rename the parameter 'study' to 'studyIdentifier'?